src/webextension/background/accounts/index.js

```diff
@@ -0,0 +1,250 @@
+        expires_at: info.expires_at || undefined,
+        id_token: info.id_token || undefined,
+      };
+      info = { ...exported };
+    }
+    return {
+      config,
+      info,
+    };
+  }
+
+  get mode() {
+    const info = this.info;
+    if (!info || !info.uid) {
+      return GUEST;
+    }
+    if (!info.refresh_token) {
+      return UNAUTHENTICATED;
+    }
+    return AUTHENTICATED;
+  }
+
+  get signedIn() { return this.mode === AUTHENTICATED; }
+
+  get uid() { return (this.info && this.info.uid) || undefined; }
+  get email() { return (this.info && this.info.email) || undefined; }
+  get keys() { return (this.info && this.info.keys) || new Map(); }
+
+  async signIn(action = "signin") {
+    let cfg = configs[this.config];
+
+    let props = {},
+        url,
+        request;
+
+    // request authorization
+    cfg = {
+      ...cfg,
+      action,
+    };
+    url = await generateAuthzURL(cfg, props);
+    let authzRsp = await browser.identity.launchWebAuthFlow({
+      url,
+      interactive: true,
+    });
+    let authzCode = processAuthzResponse(new URL(authzRsp), props);
+
+    // exchange token
+    let tokenParams = {
+      grant_type: "authorization_code",
+      code: authzCode,
+      client_id: cfg.client_id,
+    };
+    if (cfg.pkce) {
+      tokenParams.code_verifier = props.pkce;
+    } else {
+      tokenParams.client_secret = cfg.client_secret;
+    }
+    url = cfg.token_endpoint;
+    request = {
+      method: "post",
+      headers: {
+        "content-type": "application/json",
+      },
+      cache: "no-cache",
+      body: JSON.stringify(tokenParams),
+    };
+    let oauthInfo = await fetchFromEndPoint("token", url, request);
+    console.log(`oauth info == ${JSON.stringify(oauthInfo)}`);
+
+    let keys = new Map();
+    if (oauthInfo.keys_jwe) {
+      let bundle = await jose.JWE.createDecrypt(props.appKey).decrypt(oauthInfo.keys_jwe);
+      bundle = JSON.parse(new TextDecoder().decode(bundle.payload));
+      let pending = Object.keys(bundle).map(async (name) => {
+        let key = bundle[name];
+        key = await jose.JWK.asKey(key);
+        keys.set(name, key);
+      });
+      await Promise.all(pending);
+    }
+
+    // retrieve user info
+    url = cfg.userinfo_endpoint;
+    request = {
+      method: "get",
+      headers: {
+        authorization: `Bearer ${oauthInfo.access_token}`,
+      },
+      cache: "no-cache",
+    };
+    let userInfo = await fetchFromEndPoint("userinfo", url, request);
+
+    this.info = {
+      uid: userInfo.uid,
+      email: userInfo.email,
+      access_token: oauthInfo.access_token,
+      expires_at: (Date.now / 1000) + oauthInfo.expires_in,
+      refresh_token: oauthInfo.refresh_token,
+      id_token: oauthInfo.id_token,
+      keys,
+    };
+    return this;
+  }
+
+  async signOut() {
+    // TODO: implement a complete signout/forget
+    // TODO: something server side?
+    this.info = undefined;
+    return this;
+  }
+
+  status() {
```

Comment:
This name confused me a fair bit when reading code further down; I'd expect `status` to be an enum or something describing whether the user was logged in or not. Maybe something like `details`?